server/src/projects/projects.controller.ts

```diff
@@ -65,12 +65,17 @@ export class ProjectsController {
 
     try {
       if (contactId) {
-        return this.projectsService.findManyByContactId(contactId);
+        return await this.projectsService.findManyByContactId(contactId);
```

Comment:
I don't think you need to `return await` — it's the responsibility of the method consumer to await it's response.

Reply:
The `await` here is necessary to prevent rejected promises (from `findManyByContactId`) from bypassing the `catch` block below. Since `findManyByContactId` is an async function, it'll return an unhandled reject promise, which bypasses the catch block.

I wrote up a section about it in the EXCEPTIONS.md guide
https://github.com/NYCPlanning/labs-applicant-portal/pull/554/files#diff-7ec9a2cb9b2a6a0502d76ee07c26d43cR78

More about it here: https://medium.com/javascript-in-plain-english/handling-exceptions-in-javascript-f14d855be11c


I _was_ wondering if they consumer would expect a promise instead. But it appears the nest guards/intereceptors and ember adapter integrate just fine even with the await....

Not sure if there is a better approach

Reply:
Aaah. Fascinating! Great find. Thanks for clarifying... I think I've definitely had this problem before and was confounded.

It'd be like:

```javascript
  try {
    const rejectedPromise = this.aPromiseWillreject();
    return rejectedPromise;
  } catch (e) {
    // uncaught!
  }
```

Where try / catch only become promise-aware because of the `await` syntax...

We would have to mix styles to only return:

```javascript
  try {
    return this.aPromiseWillreject
      .catch(e => throw new Error(e));
  } catch (e) {
    // caught! 
  }
```

But that looks bad/is confusing.

> I _was_ wondering if they consumer would expect a promise instead.

Hm, yeah, the method caller will always get a promise return because it's an `async` function, regardless of whether `await` was used. If it wasn't an `async` function, they'd still get a promise because we know the returning method call itself returns a promise. In that case, the caller would need to try/catch it — but then we'd be in a similar spot! Gah!

I'm guessing this is something you've also dug up in practice?

Reply:
> We would have to mix styles to only return:
Exactly, yuck 😬 

> Hm, yeah, the method caller will always get a promise return because it's an async function, regardless of whether await was used. If it wasn't an async function, they'd still get a promise because we know the returning method call itself returns a promise. In that case, the caller would need to try/catch it — but then we'd be in a similar spot! Gah!

Oh right, duh me 🤦 , it's still a promise. What do you mean by dug up in practice? 

In our case, I think it's good we catch the rejected promise and re-throw it as an HttpException so that it ultimately goes through our global filter. 

Aside from the examples in the article, I haven't looked into if there are other codebases that also do this specifically, but can look around.